### PR TITLE
Unify error codes and error handling language

### PR DESCRIPTION
**Pull request by Wind4Greg**
This PR addresses issue https://github.com/w3c/vc-di-eddsa/issues/82.
It updates the document to use three main error codes (PROOF_TRANSFORMATION_ERROR, PROOF_GENERATION_ERROR, and PROOF_VERIFICATION_ERROR. ) and adds appropriate language on raising and conveying errors where appropriate.

***
<a href="https://pr-preview.s3.amazonaws.com/Wind4Greg/vc-di-eddsa/pull/84.html" title="Last updated on Jun 30, 2024, 4:15 PM UTC (401f365)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-di-eddsa/84/0fa95aa...Wind4Greg:401f365.html" title="Last updated on Jun 30, 2024, 4:15 PM UTC (401f365)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -632,8 +632,8 @@ <h4>Transformation (eddsa-rdfc-2022)</h4>
             <li>
 If <var>options</var>.<var>type</var> is not set to the string
 `DataIntegrityProof` and <var>options</var>.<var>cryptosuite</var> is not
-set to the string `eddsa-rdfc-2022` then a `PROOF_TRANSFORMATION_ERROR` MUST be
-raised.
+set to the string `eddsa-rdfc-2022` then an error MUST be raised and SHOULD
```
Comment (TallTed) [inline index.html:635]:
```suggestion
set to the string `eddsa-rdfc-2022`, an error MUST be raised and SHOULD
```

File: index.html [hunk 2]
```diff
@@ -1007,7 +1010,7 @@ <h4>Transformation (eddsa-jcs-2022)</h4>
 If <var>options</var>.<var>type</var> is not set to the string
 `DataIntegrityProof` and <var>options</var>.<var>cryptosuite</var> is not
 set to the string `eddsa-jcs-2022` then an error MUST be raised that
```
Comment (TallTed) [inline index.html:1012]:
```suggestion
set to the string `eddsa-jcs-2022`, an error MUST be raised and
```

File: index.html [hunk 3]
```diff
@@ -1007,7 +1010,7 @@ <h4>Transformation (eddsa-jcs-2022)</h4>
...
-SHOULD use the `MALFORMED_PROOF_ERROR` error code.
+SHOULD use the `PROOF_VERIFICATION_ERROR` error code.
```
Comment (TallTed) [inline index.html]:
```suggestion
SHOULD convey an error type of `PROOF_VERIFICATION_ERROR`.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Just a few, to make all these requirements read consistently.

**Review by iherman — APPROVED** [review]

**Review by dlongley — APPROVED** [review]

**Comment by Wind4Greg** [comment]
@TallTed can you look at this again. I also used "data-cite" links for the error codes per recommendation from @iherman.

**Comment by Wind4Greg** [comment]
@TallTed please check this again. I got weird merge conflicts when I added links for the error codes (to the DI spec) and don't know if I lost some of the edits.

**Review by msporny — APPROVED** [review]

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**